.github/workflows/triage.yml

```
@@ -20,42 +19,105 @@ jobs:
       - name: Retrieve the content of all columns on the board
         run: |
           gh api graphql \
-            --raw-field owner="$ORGANIZATION" \
-            --field project_number="$PROJECT_NUMBER" \
-            --raw-field repository_name="$REPOSITORY" \
+            --raw-field organization=$ORGANIZATION \
+            --field project_number=$PROJECT_NUMBER \
             --raw-field query='
-              query($owner: String!, $repository_name: String!, $project_number: Int!) {
-                repository(owner: $owner, name: $repository_name) {
-                  project(number: $project_number) {
-                    columns(first: 10) {
+              query($organization: String!, $project_number: Int!) {
+                organization(login: $organization){
+                  projectV2(number: $project_number) {
+                    id
+                    fields(first:20) {
                       nodes {
-                        id,
-                        name
+                        ... on ProjectV2Field {
+                          id
+                          name
+                        }
+                        ... on ProjectV2SingleSelectField {
+                          id
+                          name
+                          options {
+                            id
+                            name
+                          }
+                        }
                       }
                     }
                   }
                 }
               }' > project_columns.json
-          echo 'COLUMN_ID='$(jq '.data.repository.project.columns.nodes[] | select(.name == "Triage") | .id' project_columns.json) >> $GITHUB_ENV
-          echo 'COLUMN_NAME='$(jq '.data.repository.project.columns.nodes[] | select(.name == "Triage") | .name' project_columns.json) >> $GITHUB_ENV
+          echo 'PROJECT_ID='$(
+            jq \
+              '.data.organization.projectV2.id' \
+              project_columns.json
+          ) >> $GITHUB_ENV
+          echo 'STATUS_FIELD_ID='$(
```

It took me a moment to figure out what status field refers to and how it relates to columns. It became clearer after looking at the board, but I think we might still be better off renaming it so that it's clear without it:
```suggestion
          echo 'COLUMN_FIELD_ID='$(
```

Or maybe even `ID_OF_FIELD_REPRESENTING_COLUMNS`, though it's a bit too verbose IMO.

---

Yes. The `Status` is the default column field of the board layout, you can see it here: https://github.com/orgs/ethereum/projects/27/settings/fields/Status
I added a variable to customize its name as well, like you recommended here for the triage column name: https://github.com/ethereum/solidity/pull/13962#discussion_r1110010903